etna/analysis/forecast/plots.py

```diff
@@ -950,7 +949,7 @@ def plot_forecast_decomposition(
         segments = forecast_ts.segments
 
     column_names = set(forecast_ts.features)
-    components = list(match_target_components(column_names))
+    components = set(filter(lambda f: f.startswith("target_component_"), column_names))
```

Comment:
This is not how it should be done. Use `target_components_names` property of the `TSDataset`.